flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/StreamScan.scala

```diff
@@ -50,11 +51,12 @@ trait StreamScan extends CommonScan[CRow] with DataStreamRel {
       f == TimeIndicatorTypeInfo.ROWTIME_STREAM_MARKER ||
       f == TimeIndicatorTypeInfo.PROCTIME_STREAM_MARKER)
 
-    if (input.getType == cRowType && !hasTimeIndicator) {
+    if (inputType == cRowType && !hasTimeIndicator) {
```

Comment:
Hi @fhueske, thanks for your review.

I thought about your comment, and I found it's hard to come up with a test case that `inputType` here is of `CRowTypeInfo`.  Since only a datastream of these types (Row, POJO, Tuple, Case class, and Atomic Type) can be registered as a Table, the `inputType` here can never be `CRowTypeInfo`. May be we can remove the If condition here. Correct me if I am wrong.